x-pack/plugins/security_solution/public/common/components/event_details/overview/status_popover_button.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiContextMenuPanel, EuiPopover, EuiPopoverTitle } from '@elastic/eui';
+import React, { useState } from 'react';
+
+import { useAlertsActions } from '../../../../detections/components/alerts_table/timeline_actions/use_alerts_actions';
+import { Status } from '../../../../../common/detection_engine/schemas/common/schemas';
+import { CHANGE_ALERT_STATUS } from '../../../../detections/components/alerts_table/translations';
+import { FormattedFieldValue } from '../../../../timelines/components/timeline/body/renderers/formatted_field';
+import type { EnrichedFieldInfoWithValues } from '../types';
+
+interface StatusPopoverButtonProps {
+  eventId: string;
+  contextId: string;
+  enrichedFieldInfo: EnrichedFieldInfoWithValues;
+  indexName: string;
+  timelineId: string;
+  handleOnEventClosed: () => void;
+}
+
+export const StatusPopoverButton = React.memo<StatusPopoverButtonProps>(
+  ({ eventId, contextId, enrichedFieldInfo, indexName, timelineId, handleOnEventClosed }) => {
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+    const togglePopover = () => setIsPopoverOpen(!isPopoverOpen);
+    const closePopover = () => setIsPopoverOpen(false);
+    const closeAfterAction = () => {
+      closePopover();
+      handleOnEventClosed();
+    };
+
+    const { actionItems } = useAlertsActions({
+      closePopover: closeAfterAction,
+      eventId,
+      timelineId,
+      indexName,
+      alertStatus: enrichedFieldInfo.values[0] as Status,
+    });
+
+    return (
+      <EuiPopover
+        button={
+          <FormattedFieldValue
```

Comment:
consider a `useMemo` here to prevent re-renders and for consistency with most usages of the `button` prop within `security_solution`

Reply:
The `FormattedFieldValue` component ultimately renders the status badge via the `RuleStatus` component in `x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/rule_status.tsx`

To update the current default style in the screenshot below:

![default_style](https://user-images.githubusercontent.com/4459398/144661089-0fec492d-ea81-4a86-b7d7-d898d59b6af7.png)

such that it includes an `arrowDown` icon on the right-hand side of the badge, as seen in the mockup below:

![status_mock](https://user-images.githubusercontent.com/4459398/144660230-dea87541-d06a-4beb-801f-ba5a3ffca60a.png)

consider (conditionally, but hard-coded in the example below) using the `iconType="arrowDown"` and `iconSide="right"` `EuiBadge` props, e.g.:

```typescript
  const badge = (
    <StyledEuiBadge color={color} iconType="arrowDown" iconSide="right" onClick={onClick}>
      {value}
    </StyledEuiBadge>
  );
```